tidy3d/components/data/data_array.py

```diff
@@ -348,17 +360,23 @@ def interp(
             **coords_kwargs,
         )
 
+    def insert_tracers(self, tracers):
+        traced_self = self.copy(deep=False, data=tracers)
+        if isbox(tracers):
+            traced_self.attrs[AUTOGRAD_KEY] = tracers
+        return traced_self
+
     def conj(self, *args: Any, **kwargs: Any):
         """Return the complex conjugate of this DataArray."""
-        if self.tracers is not None:
-            return self.__array_wrap__(anp.conj(self.tracers))
+        if isbox(self.values.flat[0]) or AUTOGRAD_KEY in self.attrs:
```

Comment:
probably we can refactor this. or at least use `get_box` from utils? and for real as well